### PR TITLE
[WIP] add more tests

### PR DESCRIPTION
**Pull request by HolmiumTS**
#4

### CODE REVIEW

File: tests/client.rs [hunk 1]
```diff
@@ -65,35 +65,35 @@ async fn has_collection() -> Result<()> {
...
 
-//     match stub.exists().await {
-//         Ok(i) => {
-//             if !i {
-//                 panic!("Cannot find created collection.");
-//             }
-//         }
-//         Err(e) => return Err(e),
-//     };
+    match client.has_collection(NAME).await {
```
Comment (yah01) [inline tests/client.rs:86]: `assert!(client.has_collection(NAME).await?)`

File: tests/client.rs [hunk 2]
```diff
@@ -65,35 +65,35 @@ async fn has_collection() -> Result<()> {
...
 
-//     match client.drop_collection(Stub::NAME).await {
-//         Ok(()) => Ok(()),
-//         Err(e) => Err(e),
-//     }
-// }
+    match client.drop_collection(NAME).await {
+        Ok(()) => Ok(()),
+        Err(e) => Err(e),
+    }
```
Comment (yah01) [inline tests/client.rs:98]: assert the `has_collection()` returns false there

**Review by yah01 — CHANGES_REQUESTED** [review]

File: tests/client.rs [hunk 3]
```diff
@@ -65,35 +65,32 @@ async fn has_collection() -> Result<()> {
...
 }
 
-// #[tokio::test]
-// #[ignore]
-// async fn create_has_drop_collection() -> Result<()> {
-//     const URL: &str = "http://localhost:19530";
+#[tokio::test]
+async fn create_has_drop_collection() -> Result<()> {
+    const URL: &str = "http://localhost:19530";
+    const NAME: &str = "create_has_drop_collection";
```
Comment (yah01) [inline tests/client.rs:71]: testing with random name is better

File: tests/collection.rs [hunk 4]
```diff
@@ -15,55 +15,159 @@
...
-//         "result num: {}",
-//         result.first().map(|c| c.len()).unwrap_or(0),
-//     );
-
-//     Ok(())
-// }
+use rand::Rng;
+use std::collections::HashMap;
+
+async fn pre_test(collection_name: &str) -> Result<Collection> {
```
Comment (yah01) [inline tests/collection.rs:26]: `pre_test` -> `create_test_collection`

File: tests/collection.rs [hunk 5]
```diff
@@ -15,55 +15,159 @@
...
+    collection.drop().await?;
+    Ok(())
+}
+
+#[tokio::test]
+async fn collection_basic() -> Result<()> {
+    let collection = pre_test("collection_basic").await?;
+
+    let mut embed_data = Vec::<f32>::new();
+    for _ in 1..=32 * 100 {
```
Comment (yah01) [inline tests/collection.rs:53]: add a util method to create test data, this code fragment occurs often

**Review by yah01 — CHANGES_REQUESTED** [review]

File: tests/client.rs (hunk 3 above)
Comment (HolmiumTS) [reply tests/client.rs:71]: i thought using a fixed collection name may be better for debugging and locating bugs

**Pull request merged**